nilearn/input_data/masker_validation.py

```diff
@@ -0,0 +1,43 @@
+import warnings
+
+from .._utils.class_inspect import get_params
+from . import MultiNiftiMasker, NiftiMasker
+
+
+def check_embedded_nifti_masker(instance, multi=True):
+    """Base class for decomposition estimator. Takes care of mask logic and of transformation
+
+    Parameters
+    ==========
+        instance: object, instance of BaseEstimator
```

Comment:
Should you check in the code that isinstance(instance, BaseEstimator) ?

Also you could call it estimator rather than instance.